### PR TITLE
ci: update devtools-protocol automatically

### PR DESCRIPTION
**Pull request by Lightning00Blade**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -27,6 +27,7 @@ repos:
       - id: check-executables-have-shebangs
       - id: check-yaml
       - id: end-of-file-fixer
+        exclude: '.browser'
```
Comment (Lightning00Blade) [inline .pre-commit-config.yaml:30]: This way we don't need to trim this file when getting content.

File: .github/workflows/update-browser-version.yml [hunk 2]
```diff
@@ -24,15 +24,10 @@ jobs:
...
+          node-version: lts/*
           cache: npm
-      - uses: google/wireit@f3a3c79c553122e2fe5829eeac7d815326502903 # setup-github-actions-caching/v1
-      - name: Install and build npm dependencies
-        run: npm ci
-      - name: Download the latest version
-        run: |
-          browser_spec="$(npx @puppeteer/browsers install chrome@canary --path /tmp/browsers | cut -f 1 -d' ')"
-          echo "$browser_spec" > .browser
+      - name: Update browser pin and devtools-protocol
```
Comment (thiagowfx) [inline .github/workflows/update-browser-version.yml:29]:
```suggestion
      - name: Update .browser pin and devtools-protocol
```

File: .github/workflows/update-browser-version.yml [hunk 3]
```diff
@@ -24,15 +24,10 @@ jobs:
       - name: Set up Node.js
         uses: actions/setup-node@5e21ff4d9bc1a8cf6de233a3057d20ec6b3fb69d # v3.8.1
         with:
-          node-version: 18
+          node-version: lts/*
```
Comment (thiagowfx) [inline .github/workflows/update-browser-version.yml:27]: Could we leave the version unchanged in this PR?

Ideally we should use the same version in _every_ workflow. It's currently inconsistent: https://github.com/search?q=repo%3AGoogleChromeLabs%2Fchromium-bidi%20node-version&type=code

It's better to address that in a separate PR

File: tools/update_chrome_revision.mjs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+const {version, revision} = await getVersionAndRevisionForCanary();
+await updateDevToolsProtocolVersion(revision);
+
+const browserVersion = `chrome@${version}`;
+await writeFile('./.browser', browserVersion, 'utf-8');
+
+// Create new `package-lock.json` as we update devtools-protocol
+execSync('npm install --ignore-scripts');
```
Comment (thiagowfx) [inline tools/update_chrome_revision.mjs:67]: If we're still doing `npm install`, then `wireit` should be kept in the github workflow

File: .github/workflows/update-browser-version.yml (hunk 3 above)
Comment (thiagowfx) [reply .github/workflows/update-browser-version.yml:27]: https://github.com/GoogleChromeLabs/chromium-bidi/pull/1315

**Review by thiagowfx — CHANGES_REQUESTED** [review]

**Review by OrKoN — APPROVED** [review]

File: tools/update_chrome_revision.mjs (hunk 4 above)
Comment (Lightning00Blade) [reply tools/update_chrome_revision.mjs:67]: Nope that is why `--ignore-scripts` is there/

Comment (thiagowfx) [reply tools/update_chrome_revision.mjs:67]: Ah OK

File: .github/workflows/update-browser-version.yml (hunk 2 above)
Comment (Lightning00Blade) [reply .github/workflows/update-browser-version.yml:29]: I disagree as browser pin is `.browser` file. Comment is only valid if you say we should write it as `Update .browser file`.

**Review by thiagowfx — APPROVED** [review]
The only other potential comment would be to add an unit test to the node file. But given that (i) we currently do not have precedent for this (in tools/) and (ii) it would be mostly mocked anyway (because of the network requests) I'm OK with skipping this. Just make sure to test the script and the workflow.

File: .github/workflows/update-browser-version.yml (hunk 2 above)
Comment (thiagowfx) [reply .github/workflows/update-browser-version.yml:29]: Not a strong opinion here as it was just a nit, feel free to leave as is then

**Pull request merged**